frontend/tests/components/resetProjectModal.ts

```diff
@@ -0,0 +1,76 @@
+import { expect, type Download, type Locator, type Page } from '@playwright/test';
+import { BaseComponent } from './baseComponent';
+
+const PLEASE_CONFIRM_DOWNLOAD_TEXT = 'Please confirm you have downloaded a backup';
+const PLEASE_CONFIRM_PROJECT_CODE_TEXT = 'Please type the project code to confirm reset';
+const I_CONFIRM_DOWNLOAD_TEXT = 'I confirm that I have downloaded a backup';
+const DOWNLOAD_PROJECT_BACKUP_TEXT = 'Download project backup';
+const ENTER_PROJECT_CODE_TEXT = 'Enter project code to confirm';
+const PROJECT_UPLOAD_CONTROL_LABEL = 'Project zip file';
+const PROJECT_UPLOAD_BUTTON_LABEL = 'Upload Project';
+
+export class ResetProjectModal extends BaseComponent {
+  get downloadProjectBackupButton(): Locator {
+    return this.componentLocator.getByRole('link').filter({hasText: DOWNLOAD_PROJECT_BACKUP_TEXT});
+  }
+
+  get confirmBackupDownloadedCheckbox(): Locator {
+    return this.componentLocator.locator('form#reset-form').getByRole('checkbox', {name: I_CONFIRM_DOWNLOAD_TEXT});
+  }
+
+  get confirmProjectCodeInput(): Locator {
+    return this.componentLocator.locator('form#reset-form').getByLabel(ENTER_PROJECT_CODE_TEXT);
+  }
+
+  get projectUploadControl(): Locator {
+    return this.componentLocator.getByLabel(PROJECT_UPLOAD_CONTROL_LABEL);
+  }
+
+  get projectUploadButton(): Locator {
+    // return this.componentLocator.getByLabel(PROJECT_UPLOAD_BUTTON_LABEL);
+    return this.page.getByRole('button', {name: PROJECT_UPLOAD_BUTTON_LABEL});
```

Comment:
Can we do this?
```suggestion
    return this.componentLocator.getByRole('button', {name: PROJECT_UPLOAD_BUTTON_LABEL});
```

Reply:
Not sure we can, I remember switching it from componentLocator to page for some reason, but it's possible I'm thinking of a different locator. I'll check again, and make the change if componentLocator does work for this one.

Reply:
Done in commit 918c4fba05.